### PR TITLE
Integration check in

### PR DESCRIPTION
**Pull request by fleur0121**
Created integration tests, but only one test is working.

Search component
Advanced Search component <= working
HomeTest component
While writing code, I found many bugs, and some codes needed improvement, so I fixed them as well.

There are a lot of changes because of mocks, do not worry about it

### CODE REVIEW

File: tests/IntegrationTest/Spoonacular/Recommendation/HomeTest.test.jsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import { describe, test, expect } from "vitest";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import HomeTest from "src/pages/Home/HomeTest";
+import { BrowserRouter } from "react-router-dom";
+import axios from "axios";
+
```
Comment (yunahu) [inline tests/IntegrationTest/Spoonacular/Recommendation/HomeTest.test.jsx:7]: when i run it, if i import env, i dont get the env not found error anymore. maybe can you try that?

File: tests/IntegrationTest/Spoonacular/Recommendation/HomeTest.test.jsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    ];
+
+    axios.get.mockResolvedValue({
+      data: recipeMock,
+    });
+
+    // api called to fetch random recipe?
+    expect(axios.get).toHaveBeenNthCalledWith(
+      1,
+      `${env.API_URL}//spoonacular/randomRecipe`,
```
Comment (yunahu) [inline tests/IntegrationTest/Spoonacular/Recommendation/HomeTest.test.jsx:79]: Did you mean to have two / before the "spoonacular/"s?
I saw it in other lines too.  Maybe you could replace them?

File: src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx [hunk 3]
```diff
@@ -30,12 +27,12 @@ const FavoriteRecommendation = () => {
         recipes.map((recipe) =>
           Promise.all([
             axios
-              .get("http://localhost:3000/spoonacular/getRecipe", {
+              .get(`${env.API_URL}//spoonacular/getRecipe`, {
```
Comment (mgao08) [inline src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx:30]: And here, double slash

File: src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx [hunk 4]
```diff
@@ -30,12 +27,12 @@ const FavoriteRecommendation = () => {
...
                 params: { id: recipe.id, includeNutrition: true },
               })
               .then((res) => res.data),
             axios
-              .get("http://localhost:3000/spoonacular/getRecipeTaste", {
+              .get(`${env.API_URL}//spoonacular/getRecipeTaste`, {
```
Comment (mgao08) [inline src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx:35]: And here, the double slash

File: src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx [hunk 5]
```diff
@@ -15,12 +15,9 @@ const FavoriteRecommendation = () => {
     const dislikes = getDislikes();
 
     try {
-      const res = await axios.get(
-        "http://localhost:3000/spoonacular/randomRecipe",
-        {
-          params: { number: 20 },
-        }
-      );
+      const res = await axios.get(`${env.API_URL}//spoonacular/randomRecipe`, {
```
Comment (mgao08) [inline src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx:18]: I suppose there should not be double slash here too?

**Review by mgao08 — CHANGES_REQUESTED** [review]

File: src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx (hunk 5 above)
Comment (fleur0121) [reply src/pages/Home/components/FavoriteRecommendation/FavoriteRecommendation.jsx]: I've totally forgotten about this component! Thank you...!

**Review by mgao08 — APPROVED** [review]

**Pull request merged**